src/Core/Types/Types/Relay/IdFieldValueSerializer.cs

```diff
@@ -9,32 +9,34 @@
 
 namespace HotChocolate.Types.Relay
 {
-    internal sealed class FieldValueSerializer : IFieldValueSerializer
+    public class IdFieldValueSerializer : IFieldValueSerializer
     {
-        private readonly NameString _typeName;
-        private readonly IIdSerializer _innerSerializer;
-        private readonly bool _validate;
-        private readonly bool _list;
-        private readonly Type _listType;
-        private NameString _schemaName;
-
-        public FieldValueSerializer(
+        protected readonly NameString TypeName;
+        protected readonly IIdSerializer InnerSerializer;
+        protected readonly bool Validate;
+        protected readonly bool IsListType;
+        protected readonly Type ListType;
```

Comment:
```suggestion
        protected readonly Type? ListType;
```

(accompanies below change)